### ISSUE TITLE
infinite loading bug (patch-gatherings, TypeError: String.prototype.link called on null or undefined)

### ISSUE THREAD

**Issue by ahdinosaur** [ssbc/patchwork#560]
using f80296e49e5bd1b20ac5c2df3b224d7b61c2d43d

```
TypeError: String.prototype.link called on null or undefined
    at link (native)
    at ProtoSet._bind (/home/dinosaur/repos/ssbc/patchwork/node_modules/mutant/set.js:115:45)
    at ProtoSet.add (/home/dinosaur/repos/ssbc/patchwork/node_modules/mutant/set.js:51:53)
    at pull.drain (/home/dinosaur/repos/ssbc/patchwork/node_modules/patch-gatherings/gathering/obs/gathering.js:65:79)
    at /home/dinosaur/repos/ssbc/patchwork/node_modules/pull-stream/sinks/drain.js:24:37
    at /home/dinosaur/repos/ssbc/patchwork/node_modules/pull-stream/throughs/filter.js:17:11
    at callback (/home/dinosaur/repos/ssbc/patchwork/node_modules/pull-pushable/index.js:85:5)
    at Function.push (/home/dinosaur/repos/ssbc/patchwork/node_modules/pull-pushable/index.js:44:7)
    at notify (/home/dinosaur/repos/ssbc/patchwork/node_modules/pull-notify/index.js:10:20)
    at /home/dinosaur/repos/ssbc/patchwork/node_modules/pull-stream/sinks/drain.js:24:37
```

**Comment by ahdinosaur** [comment]
happens in [`patchbay`](https://github.com/ssbc/patchbay) as well

**Comment by mmckegg** [comment]
Yeah, it's everywhere!!! HAHA WHOOPS!

It's some dodgy message that I just published.

**Comment by ahdinosaur** [comment]
which message is the dodgy one?

@pietgeursen

**Comment by mmckegg** [comment]
Trying to figure that out. I've been rewriting gatherings to use ssb-about.

Must have done something bad. But it's working in my rewrite, so there's that.

Just about to publish it.

**Comment by mmckegg** [comment]
Found it:

```
{
  "key": "%Tnza+8RaPk3Mu3Vd8LMhhjqxc2j20CSrYHZq9re/jfY=.sha256",
  "value": {
    "previous": "%JGUwaoeN6IQHkiXP8HUxLEiJvEqNVWdMPemxaOkPxF0=.sha256",
    "author": "@FbGoHeEcePDG3Evemrc+hm+S77cXKf8BRQgkYinJggg=.ed25519",
    "sequence": 1913,
    "timestamp": 1497703400305,
    "hash": "sha256",
    "content": {
      "type": "about",
      "about": "%JGUwaoeN6IQHkiXP8HUxLEiJvEqNVWdMPemxaOkPxF0=.sha256",
      "image": "&yAEJ0AU2HWNzpXBOSX36PjG27ybPgeY3AQBKRfM+3wU=.sha256",
      "startDateTime": {
        "tz": "Pacific/Auckland",
        "valid": true,
        "bias": -720,
        "epoch": 1498694400000
      },
      "title": "Test Gathering",
      "description": "A test gathering"
    },
    "signature": "K7bKpUfO3K/a5pJoMoT2khBfThAxHXKt/hEF7fkPzMYdBnhpphRDa0sqv+hXZesWBbJzTeJUYWa0k+mhRGlOCQ==.sig.ed25519"
  }
}
```

Specifies image as blob string instead of link object.

**Comment by ahdinosaur** [comment]
oh, the failing line [tries to get `image.link`](https://github.com/pietgeursen/patch-gatherings/blob/8e306ac0dd51fb2d12af9e03599ea6ac7da0c0ab/gathering/obs/gathering.js#L65), so is it `%Tnza+8RaPk3Mu3Vd8LMhhjqxc2j20CSrYHZq9re/jfY=.sha256`?

i'd say this is an error in `patch-gatherings`, since Patchwork already uses a string for `image` in normal about messages. see also [`ssb-msg-schemas`](https://github.com/ssbc/ssb-msg-schemas/blob/f2cb42a5e08c7b26b2276750c76bca06b6d340ae/about.js#L43-L63)

**Comment by mmckegg** [comment]
yeah, definitely!

**Comment by mmckegg** [comment]
Everyone that has me as a FOAF and is using Patchwork v3.4.0 is going to get the infinite loading screen of death 😢 

I guess we'd better push out Patchwork v3.5.0 sooner rather than later. Or people can downgrade to an older version.

**Comment by mmckegg** [comment]
Decided to go ahead and publish v3.5.0! Have macOS and windows builds up.

@ahdinosaur can you build for linux and then untick "prerelease"?

**Comment by mmckegg** [comment]
Also you might want to pull the latest patchwork-builder and `npm install`

**Comment by ahdinosaur** [comment]
@mmckegg okay will do. having a problem with `sodium-native`, will see if i can do something similar to https://github.com/ssbc/patchwork-builder/commit/116546d0b88c8ba93e695eca6ea2d7addc9a490d to get it working.

**Comment by mmckegg** [comment]
@ahdinosaur oh yeah, I didn't even notice but looks like the macOS version has the same problem 😞

**Comment by mmckegg** [comment]
@ahdinosaur I think that 45be5c0aa4d3e5e2d1bc068c8d5666fd2eef8d78 should fix this!

**Comment by ahdinosaur** [comment]
yes, thanks! uploading meow. :cat2:

**Comment by ahdinosaur** [comment]
released! [v3.5.1](https://github.com/ssbc/patchwork/releases/tag/v3.5.1)
